### PR TITLE
Added fr localization

### PR DESCRIPTION
**Pull request by yanickouellet**
I added localization using nuxtjs-i18n.

### CODE REVIEW

File: assets/products-fr.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
+export interface Product {
```
Comment (dominic-blain) [inline assets/products-fr.ts:1]: are these duplicated interfaces from `products.ts` necessary ?

File: pages/index.vue [hunk 2]
```diff
@@ -29,17 +31,21 @@ export default Vue.extend({
...
     return {
-      title: "ReplicaZone - Snipcart Demo",
+      title: this.$i18n.t('full_title').toString()
     };
   },
-  async asyncData({ route }) {
+  async asyncData() {
     return {
-      products: Products,
+      products: {}
```
Comment (dominic-blain) [inline pages/index.vue:39]: You can access i18n in the context provided by nuxt to asyncData
```
asyncData({ i18n }) {
  return {
    products: i18n.locale === 'fr' ? ProductsFR : Products
  }
}
```

File: pages/index.vue [hunk 3]
```diff
@@ -13,12 +13,14 @@
 
 <script lang="ts">
 import Vue from "vue";
+import i18n from 'vue-i18n';
```
Comment (dominic-blain) [inline pages/index.vue:16]: This import should not be necessary AFAIK

File: pages/index.vue [hunk 4]
```diff
@@ -29,17 +31,21 @@ export default Vue.extend({
...
-  async asyncData({ route }) {
+  async asyncData() {
     return {
-      products: Products,
+      products: {}
     };
   },
+
+  mounted() {
+    this.$data.products = this.$i18n.locale == 'fr' ? ProductsFr : Products
```
Comment (dominic-blain) [inline pages/index.vue:44]: You can use the `onLanguageSwitched` callback to change products dynamically everytime the language is changed https://i18n.nuxtjs.org/callbacks#onlanguageswitched

**Review by dominic-blain — CHANGES_REQUESTED** [review]
We will probably want a language switcher button? Not too complicated to do with this nuxt module https://i18n.nuxtjs.org/lang-switcher

File: pages/index.vue (hunk 2 above)
Comment (yanickouellet) [reply pages/index.vue:39]: Didn't know, thanks!

**Comment by yanickouellet** [comment]
I checked with Emilie and we only need to send the french website to the french demo.

File: pages/index.vue (hunk 3 above)
Comment (yanickouellet) [reply pages/index.vue]: Yeah, I added it while trying something and forgot to remove it.

**Review by dominic-blain — APPROVED** [review]

**Pull request merged**